### DISCUSSION TITLE
Onshore discon question

### DISCUSSION THREAD

**Discussion by icnosvbovwb** [OpenFAST/openfast#1354] [Q&A] [answered]
Hello everyone
When I analyzed NRELOffshrBsline5MW wind turbine, the Inflowwind input file is wind90m_ 12mps_ twr.bts, which can normally use discon.dll. However, when I change WindType=1, HWindSpeed=10.1m/s, onshore does not work properly. It will make some errors.
![image](https://user-images.githubusercontent.com/107483730/204216780-2907068d-be13-49a1-9b59-f491e1069b18.png)

**Comment by jjonkman** [comment]
Dear @icnosvbovwb,

When you changed the wind speed, did you also change the initial rotor speed and initial blade-pitch angle?  If not, the controller may not respond properly.  It is generally recommend to set the initial rotor speed and initial blade-pitch angles to their expected (mean) values for the given hub-height wind speed.

Best regards,

**Comment by icnosvbovwb** [comment]
Dear @jjonkman 
![image](https://user-images.githubusercontent.com/107483730/204292056-07304bfe-0be2-4dbd-9a4a-17fd21fe7467.png)
I input different steady wind speeds, and other conditions remain unchanged as shown in the figure, the initial rotor speed and initial blade-pitch angles are 12.1rpm  and 0  deg respectively. I change the time steps to smaller, such as 0.00625, which can be calculated normally. But I set the wind speed as 10.2m/s, and 11.4m/s, which can not be calculated normally. Or is there a method to input step wind speed? For example, 3m/s, 3.1m/s, 3.2m/s to 25m/s.

**Comment by icnosvbovwb** [comment]
Dear @jjonkman 
![image](https://user-images.githubusercontent.com/107483730/204296730-ff377e99-067d-4110-a789-5d3e1c98d667.png)
![image](https://user-images.githubusercontent.com/107483730/204296899-0a8abde4-984a-46a7-9ed4-65ef5b99a548.png)
I found a small mistake, that is, the initial displacement at the front and back of the tower top(TTDspFA) should be set to 0. I set the time step(DT) to 0.025s, and the total time(TMax) is 500s. After I corrected, I found that the electric energy(GenPwr) generated by 11.4m/s steady wind is not 5MW, but about 4.7MW. I don't know where have a mistake. Please help me.

**Reply by jjonkman** [reply]
Dear @icnosvbovwb,

The generator power will depend on which aerodynamic features have been enabled in AeroDyn and may differ a bit between different releases of AeroDyn and OpenFAST.

Best regards,

**Comment by icnosvbovwb** [comment]
Dear @jjonkman 
I use the openfast V3.1 and the initial DISCON controller in your paper to apply to the onshore wind turbine. Does this mean that I cannot get the rated power at 11.4m/s steady wind speed？ 
Another question is how do I get the Blade root flexible motion (Nm) and the Rotor electrical power coefficient (-)
These parameters is not reflected in the outlist.

**Reply by jjonkman** [reply]
Dear @icnosvbovwb,

> Does this mean that I cannot get the rated power at 11.4m/s steady wind speed？

I'm not saying that.  I'm just saying that the generator power will depend on which aerodynamic features have been enabled in AeroDyn and may differ a bit between different releases of AeroDyn and OpenFAST.

I'm not sure what you mean when you ask about "blade root flexible motion (Nm)".  Are you referring to the blade-root bending and pitching moments?  These are available outputs from the structural module (ElastoDyn), when enabled.

The rotor electrical power coefficient is not an output, but you can calculate it by dividing the electrical power output (ServoDyn output `GenPwr`) by `1/2 * AirDens * RtArea * Vxh^3`, where `RtArea` is an output of AeroDyn and `Vxh` is the wind speed projected along the rotor axis.

Best regards,

**Comment by icnosvbovwb** [comment, marked as answer]
Dear @jjonkman 
Thank you for your reply. I have solved this problem.
